.github/workflows/npmjs-publish.yaml

```diff
@@ -0,0 +1,150 @@
+            CURRENT_DAY=$(date +'%Y%m%d')
+            SHORT_SHA1=$(git rev-parse --short HEAD)
+            PODMAN_DESKTOP_VERSION=$(jq -r '.version' package.json)
+            STRIPPED_VERSION=${PODMAN_DESKTOP_VERSION%-next}
+            TAG_PATTERN=${STRIPPED_VERSION}-next.$(date +'%Y%m%d%H%M')-${SHORT_SHA1}
+            GITHUB_TAG="v${TAG_PATTERN}"
+
+            echo "Computed githubTag=${GITHUB_TAG}"
+
+            echo "version=${TAG_PATTERN}" >> "${GITHUB_OUTPUT}"
+            echo "dist-tag=next" >> "${GITHUB_OUTPUT}"
+            echo "github-tag=${GITHUB_TAG}" >> "${GITHUB_OUTPUT}"
+            echo "is-tag=false" >> "${GITHUB_OUTPUT}"
+          fi
+
+  publish:
+    name: Publish to npm
+    needs: [prepare-version]
+    runs-on: ubuntu-24.04
+    # For tag releases, require an environment (so people can approve it)
+    environment: ${{ needs.prepare-version.outputs.is-tag == 'true' && 'npmjs-publication' || '' }}
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - uses: pnpm/action-setup@41ff72655975bd51cab0327fa583b6e92b6d3061 # v4.2.0
+        name: Install pnpm
+        with:
+          run_install: false
+
+      - uses: actions/setup-node@2028fbc5c25fe9cf00d9f06a71cc4710d4507903 # v6.0.0
+        with:
+          node-version: 24 # 24.x for npm publish with OIDC
+          cache: 'pnpm'
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Show publish parameters
+        run: |
+          echo "Publishing version: ${{ needs.prepare-version.outputs.version }}"
+          echo "Using dist-tag:     ${{ needs.prepare-version.outputs.dist-tag }}"
+          echo "GitHub tag:         ${{ needs.prepare-version.outputs.github-tag }}"
+          echo "Is tag ref:         ${{ needs.prepare-version.outputs.is-tag }}"
+
+      - name: Update versions in package.json files
+        run: |
+          VERSION="${{ needs.prepare-version.outputs.version }}"
+          echo "Setting version to ${VERSION}"
+
+          sed -i  "s#\"version\": \"\(.*\)\",#\"version\": \"${VERSION}\",#g" packages/extension-api/package.json
+          sed -i  "s#\"version\": \"\(.*\)\",#\"version\": \"${VERSION}\",#g" packages/webview-api/package.json
+          sed -i  "s#\"version\": \"\(.*\)\",#\"version\": \"${VERSION}\",#g" packages/ui/package.json
+          sed -i  "s#\"version\": \"\(.*\)\",#\"version\": \"${VERSION}\",#g" tests/playwright/package.json
+          sed -i  "s#\"version\": \"\(.*\)\",#\"version\": \"${VERSION}\",#g" extensions/podman/packages/api/package.json
+          sed -i  "s#\"version\": \"\(.*\)\",#\"version\": \"${VERSION}\",#g" extensions/docker/packages/api/package.json
+
+      - name: Publish Extension API to npmjs
+        run: |
+          cd packages/extension-api
+          pnpm publish --provenance --tag ${{ needs.prepare-version.outputs.dist-tag }} --no-git-checks --access public
```

Comment:
Is `--provenance` still needed?

From https://github.blog/changelog/2025-07-31-npm-trusted-publishing-with-oidc-is-generally-available/:
```
With trusted publishing, you can now:

Get automatic provenance: When using trusted publishing, npm CLI publishes provenance attestations by default. The --provenance flag is no longer needed.
```